drivers/hmis/app/graphql/mutations/submit_assessment.rb

```diff
@@ -8,34 +8,47 @@ class SubmitAssessment < BaseMutation
 
     def resolve(input:)
       assessment, errors = input.find_or_create_assessment
-      return { assessment: nil, errors: errors } if errors.any?
+      return { errors: errors } if errors.any?
 
       definition = assessment.assessment_detail.definition
       enrollment = assessment.enrollment
 
-      # Determine the Assessment Date (same as Information Date) and validate it
+      # If this is an HoH Exit, check special restrictions
+      new_hoh_enrollment = nil
+      if enrollment.head_of_household? && assessment.exit?
+        open_enrollments = Hmis::Hud::Enrollment.open_on_date.
+          where(household_id: enrollment.household_id).
+          where.not(id: enrollment.id)
```

Comment:
Pretty sure we need to check `data_source_id` since HouseholdID could be duplicated across different data sources.

Reply:
Oh good catch, thanks.

Reply:
We had several places where we made this assumption. I think I got all of them with https://github.com/greenriver/hmis-warehouse/pull/2838/commits/957b98e17e3816b73cc037193fe0ba639a67cb49